app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -312,20 +308,21 @@ class EventDetailsFragment : Fragment() {
     }
 
     private fun loadTicketFragment() {
-        // Initialise Ticket Fragment
-        val ticketFragment = TicketsFragment()
-        val bundle = Bundle()
-        bundle.putLong("EVENT_ID", eventId)
-        bundle.putString(CURRENCY, currency)
-        ticketFragment.arguments = bundle
-        findNavController(rootView).navigate(R.id.ticketsFragment, bundle, getAnimSlide())
+        TicketsFragmentArgs.Builder()
+            .setEVENTID(safeArgs.eventid)
+            .setCURRENCY(currency)
+            .build()
+            .toBundle()
+            .also { bundle ->
+                findNavController(rootView).navigate(R.id.ticketsFragment, bundle, getAnimSlide())
```

Comment:
Why are passing bundle if you are getting event id from safe args in ticket fragment?

Reply:
Because `findNavController().navigate()` call does not accept safe-args directly. You can not pass an instance of `TicketFragmentArgs` directly to it. You have to create safe args using the builder, convert that builder to a bundle, and then pass that bundle to the navigate call.

Here's the error I get when I try to pass an instance of TicketFragmentArgs directly without converting it to a bundle first:
![screenshot 2019-03-01 at 3 15 05 pm](https://user-images.githubusercontent.com/24315306/53630137-6233b580-3c35-11e9-9d01-007eb816ee13.png)


If you want to pass arguments without using a bundle, then you have to encapsulate them in a NavDirections object. Our project does not use Actions from the Nav AAC, so we don't have generated NavDirections classes for any fragments.

I feel like I am unable to understand what seems to be the issue here. I apologize if I am still wrong, can you please explain in detail to me the method you are suggesting?

Reply:
Just use `.build().toBundle()`.

Reply:
Can't we get eventId directly from the generated class in tickets fragment?

Reply:
> Just use `.build().toBundle()`.

That is kind of exactly what I am doing right now, except that instead of storing the value returned by the `toBundle()` call in a variable, I use Kotlin's `also` function to directly pass that value to the `navigate` call.

> Can't we get eventId directly from the generated class in tickets fragment?

That's exactly what we are doing right now. The `by navArgs()` delegate takes the received arguments for the fragment and type-casts them to the corresponding generated Args class (which is TicketsFragmentArgs in this case). 

SafeArgs are not a replacement for bundles, they are just a way to introduce type safety to them. Under the hood they still use bundles, but they help in verification of them to ensure that requested arguments are present inside them.